website/content/en/docs/reference/threat-model.md

```diff
@@ -29,92 +28,161 @@ A Cluster Developer has the ability to create pods via `Deployments`, `ReplicaSe
 
-**Background**: EC2 instances can exist in an AWS account outside of the Kubernetes cluster.
+**Mitigation:** Cluster Operators must enumerate the roles in the resource section of the IAM policy granted to the Karpenter role for the `iam:PassRole` action. Karpenter will fail to generate an instance profile if role that is specified in the `spec.role` section of the `EC2NodeClass` is not enumerated in the `iam:PassRole` permission.
 
-**Threat**: An actor who obtains control of the Karpenter pod’s IAM role may be able to create or terminate EC2 instances not part of the Kubernetes cluster managed by Karpenter.
+#### Execution of Arbitrary UserData
 
-**Mitigation**: Karpenter creates instances with tags, several of which are enforced in the IAM policy granted to the Karpenter IAM role that restrict the instances Karpenter can terminate. One tag requires that the instance was provisioned by a Karpenter controller (`karpenter.sh/nodepool`), another tag can include a cluster name to mitigate any termination between two clusters with Karpenter in the same account (`kubernetes.io/cluster/${CLUSTER_NAME}`. Cluster Operators also can restrict the region to prevent two clusters in the same account with the same name in different regions.
+**Threat:** Karpenter allows users to specify custom UserData as part of the `EC2NodeClass` resource. This UserData may take two shapes: a MIME multi-part archive or a NodeConfig yaml object. Karpenter does not perform any validation on this input, other than parsing the MIME multi-part archive if a header is detected. This custom UserData is then merged with UserData generated by Karpenter and is included in the Launch Template used to launch Karpenter managed nodes. Users with EDIT permissions on `EC2NodeClass` are able arbitrarily execute code on nodes launched by Karpenter.
 
-Additionally, Karpenter does not allow tags to be modified on instances unowned by Karpenter after creation, except for the `Name` and `karpenter.sh/nodeclaim` tags. Though these tags can be changed after instance creation, `aws:ResourceTag` conditions enforce that the Karpenter controller is only able to change these tags on instances that it already owns, enforced through the `karpenter.sh/nodepool` and `kubernetes.io/cluster/${CLUSTER_NAME}` tags.
+**Mitigation:** Access to `EC2NodeClass` is treated as a privileged permission. Access to the `EC2NodeClass` enables a user to not only specify custom UserData but also modify the IAM role used by the instance profile generated for the node, select the Node’s subnets and security groups, and other privileged settings. By default, only the Cluster Admin has access to this resource. If the Cluster Admin grants another user permission, this is the Cluster Admin’s responsibility. 
```

Comment:
Can we also make a note that another mitigation that can be applied here is scoping down the permissions assigned to the nodeRole so that even if someone is able to execute arbitrary code, they can't do much in the AWS account?